client-types/index.d.ts

```diff
@@ -0,0 +1,37 @@
+export {};
+
+interface ExportInfo {
+	used: boolean;
+	provideInfo: boolean | null | undefined;
+	useInfo: boolean | null | undefined;
+}
+
+interface ExportsInfo {
+	[k: string]: ExportInfo & {[k: string]: ExportInfo}
```

Comment:
This can be deeply nested:

```suggestion
	[k: string]: ExportInfo & ExportsInfo
```

Not sure if typescript likes that...